SimpleRss/SimpleRss/Services/DataService.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  DataService.swift
+//  SimpleRss
+//
+//  Created by Voldem on 2/18/19.
+//  Copyright © 2019 Vladimir Koptev. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+class DataService {
+    
+    lazy var cacheContext: NSManagedObjectContext = {
+        return appDelegate.persistentContainer.newBackgroundContext()
+    }()
+    
+    lazy var updateContext: NSManagedObjectContext = {
+        let _updateContext = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
+        _updateContext.parent = viewContext
+        return _updateContext
+    }()
+    
+    private lazy var appDelegate = {
+        return UIApplication.shared.delegate as! AppDelegate
+    }()
+    
+    private var viewContext: NSManagedObjectContext {
+        return appDelegate.persistentContainer.viewContext
+    }
+    
+    func getTopics() -> [TopicModel]? {
+        
+        let fetch = NSFetchRequest<NSFetchRequestResult>(entityName: "Topic")
+        let topics = try? updateContext.fetch(fetch)
```

Comment:
Such context methods must be called on its queue.
Check methods `perfrorm` and `performAndWait` in NSManagedObjectContext.

Also enable core data debugging `-com.apple.CoreData.ConcurrencyDebug 1`.